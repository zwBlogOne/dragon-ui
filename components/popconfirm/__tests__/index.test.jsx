import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Popconfirm from '../index';
import Button from '../../button';

describe('PopConfirm', () => {
  it('renders PopConfirm correctly', () => {
    const wrapper = render(
      <div>
        <Popconfirm content="确认删除吗" direction="top">
          <Button>Delete</Button>
        </Popconfirm>
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('behaves correctly when clicking onOk', () => {
    const onOk = jest.fn();
    const wrapper = mount(
      <div>
        <Popconfirm content="确认删除吗" direction="top" onOk={onOk}>
          Delete
        </Popconfirm>
      </div>
    );

    wrapper.find('.ui-button').at(1).simulate('click');

    expect(onOk).toHaveBeenCalled();
  });

  it('behaves correctly when clicking onCancel', () => {
    const onCancel = jest.fn();
    const wrapper = mount(
      <div>
        <Popconfirm content="确认删除吗" direction="top" onCancel={onCancel}>
          Delete
        </Popconfirm>
      </div>
    );

    wrapper.find('.ui-button').at(0).simulate('click');

    expect(onCancel).toHaveBeenCalled();
  });

  it('behaves correctly when toggling from hidden to visible', () => {
    const wrapper = mount(
      <Popconfirm content="确认删除吗" direction="top">
        Delete
      </Popconfirm>
    );

    expect(wrapper.find('.ui-popover-content').hasClass('ui-popover-content-show')).toBeFalsy();
    wrapper.setProps({ visible: true });
    expect(wrapper.find('.ui-popover-content').hasClass('ui-popover-content-show')).toBeTruthy();
  });
});
