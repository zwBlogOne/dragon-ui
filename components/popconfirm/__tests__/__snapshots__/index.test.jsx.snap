// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopConfirm renders PopConfirm correctly 1`] = `
<div>
  <div
    class="ui-popover"
  >
    <div
      class="ui-popover-content ui-popover-content-radius"
    >
      <div>
        确认删除吗
        <div
          class="ui-popover-actions"
        >
          <button
            class="ui-button theme-default"
            type="button"
          >
            取消
          </button>
          <button
            class="ui-button theme-info"
            type="button"
          >
            确定
          </button>
        </div>
      </div>
      <span
        class="ui-popover-arrow"
      />
    </div>
    <button
      class="ui-button theme-default"
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
